core/base-service/base-static.js

```diff
@@ -1,58 +1,29 @@
-import makeBadge from '../../badge-maker/lib/make-badge.js'
 import BaseService from './base.js'
 import {
   serverHasBeenUpSinceResourceCached,
   setCacheHeadersForStaticResource,
 } from './cache-headers.js'
-import { makeSend } from './legacy-result-sender.js'
-import { MetricHelper } from './metric-helper.js'
-import coalesceBadge from './coalesce-badge.js'
-import { prepareRoute, namedParamsForMatch } from './route.js'
+import { prepareRoute } from './route.js'
 
 export default class BaseStaticService extends BaseService {
-  static register({ camp, metricInstance }, serviceConfig) {
-    const { regex, captureNames } = prepareRoute(this.route)
-
-    const metricHelper = MetricHelper.create({
-      metricInstance,
-      ServiceClass: this,
-    })
-
-    camp.route(regex, async (queryParams, match, end, ask) => {
-      if (serverHasBeenUpSinceResourceCached(ask.req)) {
-        // Send Not Modified.
-        ask.res.statusCode = 304
-        ask.res.end()
-        return
-      }
-
-      const metricHandle = metricHelper.startRequest()
-
-      const namedParams = namedParamsForMatch(captureNames, match, this)
-      const serviceData = await this.invoke(
-        {},
-        serviceConfig,
-        namedParams,
-        queryParams
-      )
-
-      const badgeData = coalesceBadge(
-        queryParams,
-        serviceData,
-        this.defaultBadgeData,
-        this
-      )
-
-      // The final capture group is the extension.
-      const format = (match.slice(-1)[0] || '.svg').replace(/^\./, '')
-      badgeData.format = format
-
-      setCacheHeadersForStaticResource(ask.res)
-
-      const svg = makeBadge(badgeData)
-      makeSend(format, ask.res, end)(svg)
+  static _applyCacheHeaders({ res }) {
+    setCacheHeadersForStaticResource(res)
+  }
 
-      metricHandle.noteResponseSent()
-    })
+  static register({ app, ...serviceContext }, serviceConfig) {
+    const { regex } = prepareRoute(this.route)
+    app.get(
+      regex,
+      (req, res, next) => {
+        if (serverHasBeenUpSinceResourceCached(req)) {
+          // Send Not Modified.
+          res.status(304)
+          res.end()
+        } else {
+          next()
+        }
+      },
+      this.makeExpressHandler(serviceContext, serviceConfig)
```

Comment:
Handle the 304's using a middleware, then delegate to the base service request handler, overriding the cache headers via `_applyCacheHeaders()`. Way DRYer and I think a big improvement, though I could imagine making this a little cleaner in the future.